deepmd/dpmodel/model/linear_model.py

```diff
@@ -0,0 +1,234 @@
+# SPDX-License-Identifier: LGPL-3.0-or-later
+from typing import (
+    Dict,
+    Optional,
+)
+
+import numpy as np
+
+from deepmd.dpmodel import (
+    FittingOutputDef,
+    OutputVariableDef,
+)
+from deepmd.dpmodel.utils.nlist import (
+    build_multiple_neighbor_list,
+)
+
+from .base_atomic_model import (
+    BaseAtomicModel,
+)
+from .dp_atomic_model import (
+    DPAtomicModel,
+)
+from .pair_tab_model import (
+    PairTabModel,
+)
+
+
+class LinearModel(BaseAtomicModel):
+    """Model linearly combine a list of AtomicModels.
+
+    Parameters
+    ----------
+    models
+            This linear model should take a DPAtomicModel and a PairTable model.
+    """
```

Comment:
Why is it called `LinearModel`? I don't see it as "linear".

Reply:
Just to match the tf side, a linear combination of two models.

Reply:
The linear combination should accept any number of models, not only two. The weight can be any constant. Users have requested features including #2724, #2780, and #3099.
What you implement is more like interpolation but can be a particular case of linear combination.